src/benchmarks/micro/corefx/System.Text.Json/Utf8JsonReader/Utf8JsonReader.MultiSegment.Tests.cs

```diff
@@ -0,0 +1,283 @@
+        private ReadOnlySequence<byte> s_singleLongSingleLineCommentSequence1;
+
+        private ReadOnlySequence<byte> s_singleEmptyMultiLineCommentSequence100;
+        private ReadOnlySequence<byte> s_multipleEmptyMultiLineCommentSequence100;
+        private ReadOnlySequence<byte> s_singleShortMultiLineCommentSequence100;
+        private ReadOnlySequence<byte> s_multipleShortMultiLineCommentSequence100;
+        private ReadOnlySequence<byte> s_singleLongMultiLineComment1Sequence100;
+        private ReadOnlySequence<byte> s_singleLongMultiLineComment2Sequence100;
+
+        private ReadOnlySequence<byte> s_singleEmptySingleLineCommentSequence100;
+        private ReadOnlySequence<byte> s_multipleEmptySingleLineCommentSequence100;
+        private ReadOnlySequence<byte> s_singleShortSingleLineCommentSequence100;
+        private ReadOnlySequence<byte> s_multipleShortSingleLineCommentSequence100;
+        private ReadOnlySequence<byte> s_singleLongSingleLineCommentSequence100;
+
+        private void MultiSegmentSetup()
+        {
+            // called within SingleSegmentSetup
+
+            s_singleEmptyMultiLineCommentSequence1 = GetSequence(s_singleEmptyMultiLineComment, 1);
+            s_multipleEmptyMultiLineCommentSequence1 = GetSequence(s_multipleEmptyMultiLineComment, 1);
+            s_singleShortMultiLineCommentSequence1 = GetSequence(s_singleShortMultiLineComment, 1);
+            s_multipleShortMultiLineCommentSequence1 = GetSequence(s_multipleShortMultiLineComment, 1);
+            s_singleLongMultiLineComment1Sequence1 = GetSequence(s_singleLongMultiLineComment1, 1);
+            s_singleLongMultiLineComment2Sequence1 = GetSequence(s_singleLongMultiLineComment2, 1);
+
+            s_singleEmptySingleLineCommentSequence1 = GetSequence(s_singleEmptySingleLineComment, 1);
+            s_multipleEmptySingleLineCommentSequence1 = GetSequence(s_multipleEmptySingleLineComment, 1);
+            s_singleShortSingleLineCommentSequence1 = GetSequence(s_singleShortSingleLineComment, 1);
+            s_multipleShortSingleLineCommentSequence1 = GetSequence(s_multipleShortSingleLineComment, 1);
+            s_singleLongSingleLineCommentSequence1 = GetSequence(s_singleLongSingleLineComment, 1);
+
+            s_singleEmptyMultiLineCommentSequence100 = GetSequence(s_singleEmptyMultiLineComment, 100);
+            s_multipleEmptyMultiLineCommentSequence100 = GetSequence(s_multipleEmptyMultiLineComment, 100);
+            s_singleShortMultiLineCommentSequence100 = GetSequence(s_singleShortMultiLineComment, 100);
+            s_multipleShortMultiLineCommentSequence100 = GetSequence(s_multipleShortMultiLineComment, 100);
+            s_singleLongMultiLineComment1Sequence100 = GetSequence(s_singleLongMultiLineComment1, 100);
+            s_singleLongMultiLineComment2Sequence100 = GetSequence(s_singleLongMultiLineComment2, 100);
+
+            s_singleEmptySingleLineCommentSequence100 = GetSequence(s_singleEmptySingleLineComment, 100);
```

Comment:
What are we getting from the 1 vs 100 segment split? Should we instead be more intentional and just pick the location where the split occurs (i.e. within a comment rather than somewhere outside of it)?

Reply:
for me interesting cases are:
- many segments, segment iteration perf
- 1 or few segments

I think 1 and 100 achieve this reasonably well (this could use Params though which I didn't know it existed)

Reply:
> many segments, segment iteration perf

I don't see measuring the perf of many small-sized segments being relevant at all. If we were to regress this case in order to optimize something else (like say reasonably sized chunks of ~4K get faster), I would make that trade off. Hence, us tracking and worrying about the perf of small segments seems unnecessary to me and send the wrong signal. We shouldn't be optimizing for this case.